microprofile/WFLY-10711_health_check_extension.adoc

```diff
@@ -0,0 +1,132 @@
+* Provide an HTTP endpoint for the https://github.com/eclipse/microprofile-health/blob/master/spec/src/main/asciidoc/protocol-wireformat.adoc[health wireformat].
+  It requires the presence of the WildFly https://github.com/wildfly/wildfly-capabilities/blob/master/org/wildfly/management/http-interface/capability.adoc[HTTP management interface capability]
+* The HTTP endpoint is global to the server: it aggregates health check probes from *all deployments*.
+
+== Nice-to-Have Requirements
+
+* Support drilldown of the HTTP endpoint (e.g. to be able to query the health of individual deployment, e.g. `/health/deploymentA`)
+* Provide a WildFly management API so that other extensions can define health check probes that taken into account
+  by the HTTP endpoint (this is out of scope of this proposal but the design of the new extension must not preclude to
+  provide such an API at a later point).
+
+== Non-Requirements
+
+* MicroProfile Health 1.0 only provides a "liveness" probe (as defined by https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-probes/#define-a-liveness-command[Kubernetes]).
+__A Readiness probe is not covered by this proposal__.
+
+== Implementation Plan
+
+* Add dependencies to `io.smallrye:smallrye-health` and `org.eclipse.microprofile.health:microprofile-health-api:1.0` artifacts
+* Add a new `microprofile-health-smallrye` extension to provide Health support for deployments.
+* Provide a Undertow handler that exposes the optional health HTTP endpoint `/health` on WildFly `management-http` interface.
+
+=== Subsystem description
+
+The subsystem will be named `microprofile-health-smallrye`.
+
+The subsystem contains no attributes or child resources.
+
+The subsystem defines 1 runtime operation:
+
+* `check` whose result is the DMR representation of the Health Check JSON payload.
+  The management operation outcome will always be `success`. The _outcome of the health check_ is provided
+  by the `outcome` property on the operation's `result`:
+
+----
+  /subsystem=microprofile-health-smallrye:check
+  {
+      "outcome" => "success",
+      "result" => {
+          "outcome" => "UP",
+          "checks" => [{
+              "name" => "random",
+              "state" => "UP",
+              "data" => {"foo" => "bar"}
+          }]
+      }
+  }
+----
+
+If the global outcome is `UP`, the result's `outcome` will be `UP`, otherwise it will be `DOWN`.
+
+The subsystem will install at runtime a HTTP endpoint `/health` on the `management-http` interface to provide
+the MicroProfile Health Check endpoint.
+
+----
+curl -v -L --digest  -u admin:adminpwd http://localhost:9990/health/
+< HTTP/1.1 200 OK
+< Content-Length: 79
+< Content-Type: application/json
+...
+{"outcome":"UP","checks":[{"name":"random","state":"UP","data":{"foo":"bar"}}]}
```

Comment:
This differs from https://github.com/eclipse/microprofile-health/blob/master/spec/src/main/asciidoc/protocol-wireformat.adoc#appendix-b-json-payload-specification

Reply:
idem https://github.com/eclipse/microprofile-health/blob/1.0/spec/src/main/asciidoc/protocol-wireformat.adoc#json-schema